src/pages/api/blob.ts

```diff
@@ -20,13 +27,35 @@ export const POST: APIRoute = async ({ request, url }) => {
 			recoverAddress(hashMessage(message), signature),
 		) ?? ZeroAddress
 
-	const pathname = `${eoa}/${nanoid()}` // a random UID generated by Vercel Blob will be added after this.
+	const nanoId = nanoid()
+	const pathname = `${eoa}/${nanoId}` // a random UID generated by Vercel Blob will be added after this.
 
 	const blob = await whenNotError(file, (_file) =>
 		put(pathname, _file, {
 			access: 'public',
+			multipart: true, // To upload large files successfully.
+		}),
+	)
+
+	const client = await whenNotError(
+		new Redis({
+			url: process.env.KV_REST_API_URL,
+			token: process.env.KV_REST_API_TOKEN,
 		}),
+		async (_client) => {
+			return _client
+		},
+	)
+
+	const savedNanoId = await whenNotErrorAll(
+		[nanoId, client, blob],
+		async ([_nanoId, _client, _blob]) => {
+			return _client
+				.set(_nanoId, _blob.url)
+				.then((res) => (res ? _nanoId : new Error('Invalid res')))
+				.catch((err) => new Error(err))
+		},
 	)
 
-	return new Response(json(blob))
+	return new Response(json({ savedNanoId }))
```

Comment:
Please make it return values of the `blob` object as well as random id.

Reply:
sure.